File: examples/courseman/fevuejs/src/model/enrolment.js
export default class Enrolment {
    constructor(
        id,
        student,
        courseModule,
        internalMark,
        examMark,
        finalGrade,
        finalMark,
        startDate,
        endDate
    ) {
        this.id = id;
        this.student = student;
        this.courseModule = courseModule;
        this.internalMark = internalMark;
        this.examMark = examMark;
        this.finalGrade = finalGrade;
        this.finalMark = finalMark;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    setStudent(student) {
        this.student = student;
    }
    setCourseModule(courseModule) {
        this.courseModule = courseModule;
    }
}

File: examples/courseman/msa/modules/servicestmsa/coursemgmt-service/src/main/java/org/jda/example/coursemanmsa/coursemgnt/modules/enrolment/model/Enrolment.java
//Serialisable = true
package org.jda.example.coursemanmsa.coursemgnt.modules.enrolment.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.jda.example.coursemanmsa.coursemgnt.modules.coursemodule.model.Coursemodule;
import org.jda.example.coursemanmsa.coursemgnt.modules.student.model.Student;
import org.springframework.hateoas.RepresentationModel;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;


@Getter @Setter @ToString
@Entity
@Table(name="enrolment")
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIdentityInfo(
		  generator = ObjectIdGenerators.PropertyGenerator.class, 
		  property = "id", scope = Enrolment.class)
public class Enrolment extends RepresentationModel<Enrolment> {

	@Id
	@Column(name = "id", nullable = false)
	@GeneratedValue (strategy = GenerationType.IDENTITY)
	private int id;
	@Column(name = "student_id", nullable = false)
	private String studentId;
	@Column(name = "coursemodule_id")
	private int coursemoduleId;
	@Column(name = "internalmark")
	private double internalmark;
	@Column(name="exammark")
	private double exammark;
	@Column(name = "finalgrade")
	private String finalgrade;
	
	@ManyToOne
    @JoinColumn(name="coursemodule_id", nullable=false)
	private Coursemodule coursemodule;
	
	@ManyToOne
    @JoinColumn(name="student_id", nullable=false)
	private Student student;

}